zone/client.h

```diff
@@ -706,13 +707,13 @@ class Client : public Mob
 	void SendCrystalCounts();
 
 	uint64 GetExperienceForKill(Mob *against);
-	void AddEXP(ExpSource exp_source, uint64 in_add_exp, uint8 conlevel = 0xFF, bool resexp = false);
+	void AddEXP(ExpSource exp_source, uint64 in_add_exp, uint8 conlevel = 0xFF, bool resexp = false, NPC* npc = nullptr);
```

Comment:
We're basically requiring the NPC at this point and it is a bit gross that it is at the end of a bunch of optionals. We should clean this up later.